src/main.c

```diff
@@ -1565,14 +1579,21 @@ static void on_accept(h2o_socket_t *listener, const char *err)
 
     do {
         h2o_socket_t *sock;
-        if (num_connections(0) >= conf.max_connections) {
+        if (conf.soft_connection_limit > 0) {
+            int excess_connections = num_connections(0) - conf.soft_connection_limit;
```

Comment:
Should we better divide the number with `conn.num_threads`? Otherwise I think we'd be retiring connections unfairly aggressively on the thread that hits this condition first.